lib/grok-pure.rb

```diff
@@ -172,7 +172,16 @@ def match(text)
     end
   end # def match
 
+  # Returns the matched regexp object directly for performance at the
+  # cost of usability
+  def match_direct(text)
```

Comment:
I don't like the `direct` name, but am having a hard time coming up with a better name. 

What about calling this `execute` and documenting that it returns a MatchData on success or `nil` on failure?

We can refactor later the rest of the library to use `execute(text)` instead of invoking `@regexp.match(text)` directly as they do.